File: 476. Number Complement/findComplement.js
/**
 * @param {number} num
 * @return {number}
 */
const findComplement = (num) => {
  let bitmask = 1;
  let numCopy = num;
  while (numCopy) {
    num = num ^ bitmask;
    bitmask = bitmask << 1;
    numCopy = numCopy >> 1;
  }

  return num;
};


console.log("findComplement: ", findComplement(5)); // 2

File: 476. Number Complement/findComplement.py
class Solution:
  def findComplement(self, num: int) -> int:
    bitmask = 1
    numCopy = num
    while numCopy:
      num = num ^ bitmask
      bitmask = bitmask << 1
      numCopy = numCopy >> 1

    return num


solution = Solution()
print("findComplement: ", solution.findComplement(5)) # 2
